docs/new-idp-requirements.md

```diff
@@ -0,0 +1,26 @@
+# New IDP Requirements
+
+## Summary
+
+This document describes the minimum requirements for adding a new IDP (Identity Provider) to the Sigstore Public Good Deployment.
+
+## Requirements
```

Comment:
We should also mention that we reserve the right to remove your identity provider from the public instance if it's found to cause technical issues or has security vulnerabilities.